### DISCUSSION TITLE
Hasura-storage on Google Cloud

### DISCUSSION THREAD

**Discussion by vnavalianyi** [nhost/hasura-storage#219] [Q&A]
Hello everyone, has anyone managed to run Hasura Storage on Google Cloud using Cloud Storage with [HMAC keys](https://cloud.google.com/storage/docs/authentication/hmackeys?authuser=1)? Perhaps you could provide an example of a docker-compose file with the correct ENV variables:
- `S3_ENDPOINT:`
- `S3_ACCESS_KEY:`
- `S3_SECRET_KEY:`
- `S3_BUCKET:`
- `S3_ROOT_FOLDER:`

I tried, but I am getting an error:
![image_2024-08-30_18-00-34](https://github.com/user-attachments/assets/6e3e6db7-aa17-4c54-8e78-6c24911d4459)

My current docker compose storage looks like this:
```yaml
nvp-storage:
  container_name: "nvp-storage"
  image: nhost/hasura-storage:0.6.1
  restart: unless-stopped
  ports:
    - '8000:8000'
  environment:
    DEBUG: "true"
    HASURA_METADATA: 1
    HASURA_ENDPOINT: 'http://nvp-graphql-engine:${HASURA_GRAPHQL_SERVER_PORT}/v1'
    HASURA_GRAPHQL_ADMIN_SECRET: ${HASURA_GRAPHQL_ADMIN_SECRET:-hello123}
    S3_ENDPOINT: https://storage.googleapis.com
    # S3_REGION: europe-west6
    S3_ACCESS_KEY: ${S3_ACCESS_KEY} #Google HMAC Access key
    S3_SECRET_KEY: ${S3_SECRET_KEY} #Google HMAC Secret key
    S3_BUCKET: "nvp-storage"
    # S3_BUCKET: "nvptest" #aws
    # S3_ROOT_FOLDER: "uploads"
    POSTGRES_MIGRATIONS: 1
    POSTGRES_MIGRATIONS_SOURCE: postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@10.66.190.3:5432/postgres?sslmode=disable
  command: serve
  networks:
    - nvp

**Comment by YuriyKabanenko** [comment]
I'm struggling with same problem
